app/controllers/visit_groups_controller.rb

```diff
@@ -31,7 +31,17 @@ def update
     if @visit_group.update_attributes(visit_group_params)
       render nothing: true
     else
-      render json: @visit_group.errors, status: :unprocessable_entity
+      if @visit_group.day.nil?
```

Comment:
I would add a comment explaining what this is doing, since this is a little out of the ordinary.